admin-dev/themes/new-theme/js/pages/product/edit/product-options-manager.ts

```diff
@@ -79,6 +80,28 @@ export default class ProductOptionsManager {
     });
   }
 
+  private initProductVisibilityList(): void {
+    const defaultSelectedInput = document.querySelector(`${ProductMap.options.visibilityRadio}:checked`) as HTMLInputElement;
+    const descriptionField = document.querySelector<HTMLDivElement>('.js-description');
+
+    if (descriptionField === null) {
+      return;
+    }
+
+    descriptionField.innerHTML = `${defaultSelectedInput.dataset.description}`;
+
+    const inputs: NodeListOf<HTMLInputElement> = document.querySelectorAll(ProductMap.options.visibilityRadio);
```

Comment:
```suggestion
    const inputs = document.querySelectorAll<HTMLInputElement>(ProductMap.options.visibilityRadio) ?? [];
```
You also have a generic on `querySelectorAll` this way you don't need to explicitly type `inputs` because TypeScript can infer the type accurately

The ` ?? []` at the end prevents you from checking that `inputs` is not null